File: adapter/src/main/java/com/hexadeventure/adapter/in/rest/game/movement/MovementActionDTO.java
package com.hexadeventure.adapter.in.rest.game.movement;

import com.hexadeventure.model.movement.MovementAction;

import java.util.List;

public record MovementActionDTO(int x, int y, ResourceActionDTO resource, List<EnemyMovementDTO> enemyMovements) {
    public static MovementActionDTO fromModel(MovementAction model) {
        return new MovementActionDTO(model.x(),
                                     model.y(),
                                     ResourceActionDTO.fromModel(model.resourceAction()),
                                     model.enemyMovements().stream().map(EnemyMovementDTO::fromModel).toList());
    }
}

File: hexaDEVenture-unity/Assets/Scripts/Network/In/Game/movement/MovementActionDTO.cs
using System.Collections.Generic;
using Network.In.Game.Map;

namespace Network.In.Game.movement {
    public record MovementActionDTO(
        Vector2DTO OriginalPosition,
        Vector2DTO TargetPosition,
        ResourceActionDTO Resource,
        IReadOnlyList<EnemyMovementDTO> EnemyMovements,
        bool StartCombat
    );
}